### ISSUE TITLE
[BUG] Multiple progressbars get printed when using v1.1.11, the same code does not have this issues when using v1.1.9

### ISSUE THREAD

**Issue by xplshn** [hedzr/progressbar#8]
Just changing the version from v1.1.9 to v1.1.11 broke `dbin`, so I don't think this is an issue with our code. I'm not sure what's going wrong with the library either, I think newlines are being printed. The multiple duplicate progressbars only appear once the progressbar has reached 100%, once it has reached 100%, its duplicates appear.

![image](https://github.com/user-attachments/assets/6c268984-588b-4bbf-8520-763450b183ac)

**Comment by hedzr** [comment]
In v1.1.11, multipb.go had a change at line 169, which restore the dirtyFlag to initial state so that `MultiPB` can reset itself.

> For the old release, reusing `mpb   progressbar.MultiPB` is harmless. But i'm not sure what effects will be taken in new version.

I guess that's why the matter appears in your case.

**Comment by hedzr** [comment]
Because you're reusing `mpb   progressbar.MultiPB`.
The safe usage is using progressbar.New() and Close() it on demand.

In my design, multibar2's `downloadGroups` shall be:

```go
func downloadGroups() {
	mpb := progressbar.New()
	
	for _, group := range [][]string{
		{"1.14.1", "1.15.1"},
		{"1.16.1", "1.17.1", "1.18.1"},
	} {
		doEachGroupWithTasks(mpb, group)
	}

	mpb.Close() // cleanup

	mpb = progressbar.New()

	for _, group := range [][]string{
		{"1.20.1", "1.21.1"},
		{"1.22.1"},
	} {
		doEachGroupWithTasks(mpb, group)
	}

	mpb.Close() // cleanup
}
```

But, i also understand the situation in your project. So i'm reviewing to reorganize the original thoughts.

**Comment by hedzr** [comment]
i hope 9408a83 can solve it. please try, thanks so much.

**Comment by akkuman** [comment]
> i hope [9408a83](https://github.com/hedzr/progressbar/commit/9408a83454a43ba3093bd4dd99699cb9c1ef93f7) can solve it. please try, thanks so much.

it's work!!!

**Comment by hedzr** [comment]
really a good news for me. then i`m releasing the patched version.

**Comment by hedzr** [comment]
the patched version v1.1.12 sent.
thx for ur patience.

**Comment by xplshn** [comment]
Thanks for the fixes @hedzr
